api/pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -381,6 +381,10 @@ type FelixConfigurationSpec struct {
 	// is sent directly from the remote node.  In "DSR" mode, the remote node appears to use the IP of the ingress
 	// node; this requires a permissive L2 network.  [Default: Tunnel]
 	BPFExternalServiceMode string `json:"bpfExternalServiceMode,omitempty" validate:"omitempty,bpfServiceMode"`
+	// BPFDSROptoutCIDRs is a list of CIDRs which are excluded from DSR. That is, clients
+	// in those CIDRs will accesses nodeports as if BPFExternalServiceMode was set to
+	// Tunnel.
+	BPFDSROptoutCIDRs *[]string `json:"bpfDSROptoutCIDRs,omitempty" validate:"omitempty,cidrs"`
```

Comment:
Was surprised to see the pointer to a slice (slices already have a nil value) but it seems like we do it elsewhere :thinking: . Maybe it allows for a node-specific felix config to override the global one to `nil`.  Not sure that will actually work with omitempty etc.

Reply:
yup me also surprised :man_shrugging: